### DISCUSSION TITLE
Performance issues in filters after upgrading to Nova 4.*

### DISCUSSION THREAD

**Discussion by sfroemkenjw** [laravel/nova-issues#5716] [Q&A]
- Laravel Version: 10.14.1
- Nova Version: 4.26.3
- PHP Version: 8.2.3
- Database Driver & Version: MariaDB 10.4
- Operating System and Version: MacBook Pro 16" 2019 Ventura 13.4 (22F66)
- Browser type and version: Google Chrome Version 114.0.5735.198 (Offizieller Build) (x86_64)
- Reproduction Repository: Redacted

### Description:
I just upgrade Laravel Nova from 2.1.0 to 4.26.3
In earlier version opening the filter needs less then 300ms and opening the filter for email addresses needs less than 700ms. Now, after updating to newest Nova version opening the filter box needs ~12 seconds and opening the email selectbox needs ~24 seconds. After nearly 30 seconds the browser ask's me to close the tab or to wait. Seems the handling of ~20.000 records for the selectbox has changed a lot after upgrading. I don't see any errors in Browser Console.

### Detailed steps to reproduce the issue on a fresh Nova installation:
* Install current version of Nova
* `composer req spatie/laravel-activitylog ^4.7 spatie/laravel-permission ^5.10`
* `php artisan vendor:publish --provider="Spatie\Activitylog\ActivitylogServiceProvider" --tag="activitylog-migrations"`
* `php artisan vendor:publish --provider="Spatie\Permission\PermissionServiceProvider"`
* `php artisan migrate`
* `php artisan nova:filter EmailDomainFilter`
* Fill method `options()` in our new filter with:

```
return User::query()
            ->selectRaw('SUBSTRING_INDEX(email, \'@\', -1) AS domain')
            ->groupBy('domain')
            ->orderBy('domain')
            ->pluck('domain', 'domain')
            ->toArray();
```

* Visit UserResource and change filters() method:

```
        return [
            new AccountTypeFilter(),
            new AccountStatusFilter(),
            new EmailDomainFilter(),
            new UserTypeFilter(),
            new UserGroupFilter(),
        ];
```
* Add some fields in fields() in UserResource
* Activate some default logging in UserResource

```
use LogsActivity;
...
public function getActivitylogOptions(): LogOptions
    {
        return LogOptions::defaults();
    }
```

* Fill up `users` table with ~60.000 records
* If you execute the query on DB directly it just needs 0.0793 seconds. So, we have a rendering problem.

**Comment by didovych** [comment]
We also face performance issues with filters on our project.

- Laravel Version: 10.13.5
- Nova Version: 4.32.12
- PHP Version: 8.2.13
- Browser type and version: Google Chrome Version 121.0.6167.189 (Official Build) (64-bit)

The filters are extremely slow when there are many filter options. For example, the "By city" filter contains 10000 options.

The performance analysis shows that `floating-ui` can be the reason:
![nova-filters](https://github.com/laravel/nova-issues/assets/13838681/81f233a3-3563-43d1-adbd-ca22da006976)
